server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -185,6 +185,7 @@ public Response doPost(
       @Context final HttpServletRequest req // used to get request content-type, remote address and AuthorizationInfo
   ) throws IOException
   {
+    final long startMs = System.currentTimeMillis();
```

Comment:
Also I think better to call the variable `startTimestamp` to make it harder to mess it with `startNs`, but probably doesn't matter for the backport branch.